### PR TITLE
Add FRS for SKR Pro

### PR DESCRIPTION
**Pull request by JohnnyTheOne**
Add Filament Runout Sensor

### CODE REVIEW

**Comment by Bob-the-Kuhn** [comment]
Is this generally useful or is it specific to your printer?

**Comment by sjasonsmith** [comment]
> Is this generally useful or is it specific to your printer?

This is in accordance to the board's documentation, which has E0-E2 connectors, in place of X/Y/Z MAX endstop connectors.

Unfortunately, Marlin also defines these pins as X/Y/Z_MAX, rather than for their labeled purpose as filament sensors. This introduced problems for sensorless homing, which plays tricks to swap connectors so that the MIN connector is used regardless of homing direction.

In my opinion, the board should be changed to only declare X/Y/Z_STOP, and declare the filament sensors according to the board's documentation. I don't think pins files should by default stray from the board's labeling, but at this point there is an existing userbase relying on it this way.

File: Marlin/src/pins/stm32f4/pins_BTT_SKR_PRO_common.h [hunk 1]
```diff
@@ -98,6 +98,13 @@
   #define Z_MIN_PROBE_PIN                   PA2
 #endif
 
+//
+// Filament Runout Sensor
+//
+#ifndef FIL_RUNOUT_PIN
+  #define FIL_RUNOUT_PIN                    PE15
+#endif
+
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f4/pins_BTT_SKR_PRO_common.h:113]: As long as this is being added, you should add all three.

```suggestion
#ifndef FIL_RUNOUT2_PIN
  #define FIL_RUNOUT2_PIN                    PE10
#endif
#ifndef FIL_RUNOUT3_PIN
  #define FIL_RUNOUT3_PIN                    PG5
#endif

```

**Comment by JohnnyTheOne** [comment]
Is for general use

**Comment by thinkyhead** [comment]
> Unfortunately, Marlin also defines these pins as X/Y/Z_MAX

Actually, this should not be problematic. The "STOP" pins are still defined, and the definition of the "other endstop" pins are just provided as a convenience, so they can be referred to in the configurations by name if there is a need.

**Pull request merged**